src/ui/Pager/Pager.ts

```diff
@@ -68,20 +73,35 @@ export class Pager extends Component {
     enableNavigationButton: ComponentOptions.buildBooleanOption({ defaultValue: true }),
 
     /**
-     * Specifies the maximum number of pages to display if enough results are available.
+     * This is a deprecated option. Instead, the pager will automatically adapt itself on each new query.
      *
-     * This property is typically set when the default number of accessible results from the index has been changed from
-     * its default value of `1000` (10 results per page X 100 maxNumberOfPage).
+     * This was used to set the default number of page accessible from the index.
      *
-     * Default value is `100`.
+     * Use {@link Pager.options.maximumNumberOfResultsFromIndex} instead if you modified the settings from the index.
+     *
+     * @deprecated
+     */
+    maxNumberOfPages: ComponentOptions.buildNumberOption({
```

Comment:
I'm not quite sure we should update the actual documentation of deprecated options.

We could rather add the new info next to the @deprecated tag.

So, we would have something like this:

```
Specifies the maximum number of pages to display if enough results are available.

This property is typically set when the default number of accessible results from the index has been changed from its default value of `1000`.

Default value is `100`

@deprecated This is a deprecated option. The `Pager` now automatically adapts itself on each new query, so you no longer need to specify a value for this option. However, if the default maximum number of accessible results value was changed on your Coveo index, you should use the [`maximumNumberOfResultsFromIndex`]{@link Pager.options.maximumNumberOfResultsFromIndex} option to specify the new value.
```
On the generated doc site, the "deprecated" tag, along with its text, will be displayed in bright red at the very beginning of the option documentation. Should be clear enough :)